### PR TITLE
Update check Catalyst (upload-artifact downgrade && OQC unpin)

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**
Downgrade upload-artifact to 4.3.4 and unpin OQC for Mac OS

### CODE REVIEW

**Review by erick-xanadu — APPROVED** [review]

File: .github/workflows/build-wheel-macos-arm64.yaml [hunk 1]
```diff
@@ -410,7 +410,7 @@ jobs:
     - name: Install OQC client
       if: ${{ matrix.python_version == '3.9' || matrix.python_version == '3.10'}}
```
Comment (dime10) [inline .github/workflows/build-wheel-macos-arm64.yaml:411]: Can we run this with other python versions now?

**Comment by rmoyard** [comment]
[sc-70535]

File: .github/workflows/build-wheel-macos-arm64.yaml (hunk 1 above)
Comment (rmoyard) [reply .github/workflows/build-wheel-macos-arm64.yaml:411]: I am trying that in another PR https://github.com/PennyLaneAI/catalyst/pull/989

**Comment by rmoyard** [comment]
https://github.com/actions/upload-artifact/issues/590

**Pull request merged**